examples/fluids/src/dm_utils.c

```diff
@@ -96,6 +47,199 @@ PetscErrorCode DMFieldToDSField(DM dm, DMLabel domain_label, PetscInt dm_field,
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+/**
+  @brief Create `CeedElemRestriction` from `DMPlex`.
+
+  Not collective across MPI processes.
+
+  @param[in]   ceed          `Ceed` context
+  @param[in]   dm            `DMPlex` holding mesh
+  @param[in]   domain_label  `DMLabel` for `DMPlex` domain
+  @param[in]   label_value   Stratum value
+  @param[in]   height        Height of `DMPlex` topology
+  @param[in]   dm_field      Index of `DMPlex` field
+  @param[out]  restriction   `CeedElemRestriction` for `DMPlex`
+
+  @return An error code: 0 - success, otherwise - failure
+**/
+PetscErrorCode DMPlexCeedElemRestrictionCreate(Ceed ceed, DM dm, DMLabel domain_label, PetscInt label_value, PetscInt height, PetscInt dm_field,
+                                               CeedElemRestriction *restriction) {
+  PetscInt num_elem, elem_size, num_dof, num_comp, *restriction_offsets_petsc;
+  CeedInt *restriction_offsets_ceed = NULL;
+
+  PetscFunctionBeginUser;
+  PetscCall(
+      DMPlexGetLocalOffsets(dm, domain_label, label_value, height, dm_field, &num_elem, &elem_size, &num_comp, &num_dof, &restriction_offsets_petsc));
+  PetscCall(IntArrayP2C(num_elem * elem_size, &restriction_offsets_petsc, &restriction_offsets_ceed));
+  PetscCallCeed(ceed, CeedElemRestrictionCreate(ceed, num_elem, elem_size, num_comp, 1, num_dof, CEED_MEM_HOST, CEED_COPY_VALUES,
+                                                restriction_offsets_ceed, restriction));
+  PetscCall(PetscFree(restriction_offsets_ceed));
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+/**
+  @brief Create `CeedElemRestriction` from `DMPlex` domain for mesh coordinates.
+
+  Not collective across MPI processes.
+
+  @param[in]   ceed          `Ceed` context
+  @param[in]   dm            `DMPlex` holding mesh
+  @param[in]   domain_label  Label for `DMPlex` domain
+  @param[in]   label_value   Stratum value
+  @param[in]   height        Height of `DMPlex` topology
+  @param[out]  restriction   `CeedElemRestriction` for mesh
+
+  @return An error code: 0 - success, otherwise - failure
+**/
+PetscErrorCode DMPlexCeedElemRestrictionCoordinateCreate(Ceed ceed, DM dm, DMLabel domain_label, PetscInt label_value, PetscInt height,
+                                                         CeedElemRestriction *restriction) {
+  DM dm_coord;
+
+  PetscFunctionBeginUser;
```

Comment:
What is the standard PETSc convention here? I always have a blank line after `PetscFunctionBeginUser`

Reply:
From https://petsc.org/release/developers/style/#c-formatting
> 4. From All PETSc subroutine code blocks must start with a single blank line between the local variable declarations followed by [PetscFunctionBegin](https://petsc.org/release/manualpages/Sys/PetscFunctionBegin/).

From https://petsc.org/release/developers/style/#usage-of-petsc-functions-and-macros
> 7. Do not put a blank line immediately after [`PetscFunctionBegin`](https://petsc.org/release/manualpages/Sys/PetscFunctionBegin/); or a blank line immediately before `PetscFunctionReturn(PETSC_SUCCESS);`.

So technically it's for `PetscFunctionBegin` rather `PetscFunctionBeginUser`, but I'm applying it to both.